src/button-dropdown/internal.tsx

```diff
@@ -259,7 +263,8 @@ const InternalButtonDropdown = React.forwardRef(
             position="static"
             role="menu"
             decreaseTopMargin={true}
-            ariaLabelledby={hasHeader ? headerId : undefined}
+            ariaLabel={ariaLabel}
+            ariaLabelledby={hasHeader ? headerId : shouldLabelWithTrigger ? triggerId : undefined}
```

Comment:
What is header and why trigger ARIA label is preferred over the header?

Reply:
`header` is a combination of `title` and `description`, these properties are available only internally and can be used with `variant: 'navigation'` inside TopNavigation. 

Priorities are the following: 

- if there is a header inside dropdown, than it is used to label menu list
- else if we have aria-label - use it
- else if the variant is _not_ `icon` or `inline-icon` - label by trigger text

In case of a split button variant I assume there will be an aria-label, because trigger itself is an icon and we can't auto-label dropdown with it